src/cluster.c

```diff
@@ -1300,12 +1300,27 @@ static void appendClusterNodeString(RedisModuleString *ret, char node_id[41], No
     RedisModule_FreeString(NULL, str);
 }
 
+static void appendSpecialClusterNodeString(RedisModuleString *ret, unsigned int j, char *direction, ShardGroupNode *node)
+{
+    size_t len;
+    const char *temp;
+
+    RedisModuleString *str = RedisModule_CreateStringPrintf(NULL, "[%u-%s-%.*s]\r\n", j, direction, 40, node->node_id);
```

Comment:
In Redis max length of the node_id is 40 (it's true for us too, but let's keep the same format)

Reply:
I dont understand the Q, aren't I limiting it to 40 here?